File: Bubble Sort Python/bubble.py
import time


def sortowanie_babelkowe(lista):
    n = len(lista)

    while n > 1:
        zamien = False
        for l in range(0, n-1):
            if lista[l] > lista[l+1]:
                lista[l], lista[l+1] = lista[l+1], lista[l]
                zamien = True        
        n -= 1
        if zamien == False: break
   
    
    return lista
        
nazwy_plików = []
zawartość = []

pliczki = [
    "plik1",
    "plik2",
    "plik3",
    "plik4",
    "plik5",
    "plik6",
    "plik7",
    "plik8",
    "plik9",
    "plik10",
    "plik11",
    "plik12",
    "plik13",
    "plik14",
    "plik15",
    "plik16",
    "plik17",
    "plik18",
    ]

nazwy_plików.append("liczby_losowe_100k.txt")
nazwy_plików.append("liczby_losowe_10k.txt")
nazwy_plików.append("liczby_losowe_1k.txt")
nazwy_plików.append("liczby_malejace_100k.txt")
nazwy_plików.append("liczby_losowe_10k.txt")
nazwy_plików.append("liczby_losowe_1k.txt")
nazwy_plików.append("liczby_malejaco_rosnacy_100k.txt")
nazwy_plików.append("liczby_malejaco_rosnacy_10k.txt")
nazwy_plików.append("liczby_malejaco_rosnacy_1k.txt")
nazwy_plików.append("liczby_rosnace_100k.txt")
nazwy_plików.append("liczby_rosnace_10k.txt")
nazwy_plików.append("liczby_rosnace_1k.txt")
nazwy_plików.append("liczby_rosnaco_malejacy_100k.txt")
nazwy_plików.append("liczby_rosnaco_malejacy_10k.txt")
nazwy_plików.append("liczby_rosnaco_malejacy_1k.txt")
nazwy_plików.append("liczby_stale_100k.txt")
nazwy_plików.append("liczby_stale_10k.txt")
nazwy_plików.append("liczby_stale_1k.txt")



i = 0

for i in range (18):

    pliczki[i] = open(nazwy_plików[i], "r").read()

    lines = pliczki[i].split("\n")

    for line in lines:
        zawartość.append(int(line))


    start = time.time()
    sortowanie_babelkowe(zawartość)
    stop = time.time()

    t = stop - start

    print ("Czas wykonania pliku: ")
    print (nazwy_plików[i])
    print (t)

File: Bubble Sort C++/bubble.cpp
#include <iostream>
#include <cstdlib>
#include <fstream>
#include <ctime>
#include <windows.h>
#include <algorithm>

using namespace std;

const int ilePlikow = 18;
const int ileTestow = 3;

string nazwaPliku[ilePlikow];
int x;
int i;
int j;
int k;
int l;
int m;
int ile;
int ileliczb[ilePlikow];
int dane[100000];
int daneDoSort[100000];

double pomiary[ilePlikow][ileTestow];
double start;
double koniec;
double czas;


int main()
{
    nazwaPliku[0] = "liczby_losowe_1k.txt";
    ileliczb[0] = 1000;
    nazwaPliku[1] = "liczby_losowe_10k.txt";
    ileliczb[1] = 10000;
    nazwaPliku[2] = "liczby_losowe_100k.txt";
    ileliczb[2] = 100000;
    nazwaPliku[3] = "liczby_malejace_1k.txt";
    ileliczb[3] = 1000;
    nazwaPliku[4] = "liczby_malejace_10k.txt";
    ileliczb[4] = 10000;
    nazwaPliku[5] = "liczby_malejace_100k.txt";
    ileliczb[5] = 100000;
    nazwaPliku[6] = "liczby_malejaco_rosnacy_1k.txt";
    ileliczb[6] = 1000;
    nazwaPliku[7] = "liczby_malejaco_rosnacy_10k.txt";
    ileliczb[7] = 10000;
    nazwaPliku[8] = "liczby_malejaco_rosnacy_100k.txt";
    ileliczb[8] = 100000;
    nazwaPliku[9] = "liczby_rosnace_1k.txt";
    ileliczb[9] = 1000;
    nazwaPliku[10] = "liczby_rosnace_10k.txt";
    ileliczb[10] = 10000;
    nazwaPliku[11] = "liczby_rosnace_100k.txt";
    ileliczb[11] = 100000;
    nazwaPliku[12] = "liczby_rosnaco_malejacy_1k.txt";
    ileliczb[12] = 1000;
    nazwaPliku[13] = "liczby_rosnaco_malejacy_10k.txt";
    ileliczb[13] = 10000;
    nazwaPliku[14] = "liczby_rosnaco_malejacy_100k.txt";
    ileliczb[14] = 100000;
    nazwaPliku[15] = "liczby_stale_1k.txt";
    ileliczb[15] = 1000;
    nazwaPliku[16] = "liczby_stale_10k.txt";
    ileliczb[16] = 10000;
    nazwaPliku[17] = "liczby_stale_100k.txt";
    ileliczb[17] = 100000;


    for (int i = 0; i <= ilePlikow; i++)
    {
        ifstream odczyt(nazwaPliku[i].c_str());       
        ile = ileliczb[i];
        for (j = 0; j < ile; j++)
        {
            odczyt >> dane[j];
        }
        cout << nazwaPliku[i].c_str() << endl;
        for (k = 1; k <= ileTestow; k++)                      
        {
            memcpy(daneDoSort, dane, sizeof(daneDoSort)); 

            start = clock();                                     
            for (l = 0; l < ile - 1; l++)
                for (m = 0; m < ile - 1; m++)
                    if (daneDoSort[m] > daneDoSort[m + 1]) swap(daneDoSort[m], daneDoSort[m + 1]);
            
            
            koniec = clock();                                     

            czas = (double)(koniec - start) / CLOCKS_PER_SEC;       
            cout << (double)czas << "\t";

            pomiary[i][k];                                    


        }
        cout << endl;




        odczyt.close();
    }

    return 0;
}
